tests/test_index.py

```diff
@@ -0,0 +1,21 @@
+import elasticsearch
+
+from search.index import index_document
+
+
+def test_index_document_logs_connection_errors(caplog, mocker):
+    es_index = mocker.patch('search.index.Elasticsearch.index')
+    es_index.side_effect = elasticsearch.exceptions.ConnectionError()
```

Comment:
Worth looking at pytest-socket to prevent any mistakes.

Reply:
could you elaborate please?

Reply:
Currently the tests seem to prevent actual connections being made. In the event that one's missed, we don't want real requests being made.

(Hopefully tests would still fail though; relates to the `elife-bot`, where tests rely on being able to make _real_ HTTP requests.)

Reply:
These are unit tests. Connections are made by `elasticsearch-py` api. The purpose of these tests is to make sure the code I've written works as expected.

I think integration tests should be written separately.

Reply:
Yes, that's why I'm suggesting it. (Someone writing a test doesn't patch something that results in a HTTP request being made, pytest-socket would catch it.)

Reply:
added `pytest-socket` as dev dependency and disabled sockets.